config/helm/aws-node-termination-handler/templates/_helpers.tpl

```diff
@@ -48,7 +48,6 @@ Common labels
 {{- if .Chart.AppVersion }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 {{- end }}
-app.kubernetes.io/component: {{ .Release.Name }}
```

Comment:
Just to clarify that my understanding of this change: the bug previously was that `component` was set to `aws-node-termination-handler`, which is not correct. The value should have been `daemonset` or `deployment` (depending on the way NTH runs in the cluster). Your solution is to remove this from the generic `labels` and instead assign it in the supersets of `labels` called `labelsDeployment` and `labelsDaemonset`, and then to use the unrelated `selectorLabelsDeployment` for the service monitor template, which separately sets `component` to `deployment`.

Do I understand correctly?

Reply:
Yes that is correct. When I refactored the chart I accidentally left the component in the main labels function and forgot to add the component labels functions. Without these functions the component was incorrectly set in the resource labels but the only bug would be the ServiceMonitor as that's the only selector working on these labels.